app/scripts/directives/editPvc.js

```diff
@@ -0,0 +1,75 @@
+'use strict';
+
+angular.module("openshiftConsole")
+  .directive("editPvc",
+             function($uibModal,
+                      $filter,
+                      $routeParams,
+                      APIService,
+                      DataService,
+                      ProjectsService,
+                      NotificationsService,
+                      Logger) {
+
+    return {
+      restrict: "E",
+      scope: {
+        // pvc object
+        pvc: "<",
+      },
+      template: '<a href="" ng-click="openEditModal()" role="button">Expand PVC</a>',
+      replace: true,
+      link: function(scope) {
+
+        scope.openEditModal = function() {
+          // opening the modal with settings scope as parent
+          var modalInstance = $uibModal.open({
+            templateUrl: 'views/modals/edit-resource.html',
+            controller: 'EditModalController',
+            scope: scope
+          });
+
+          var hideErrorNotifications = function() {
+            NotificationsService.hideNotification("expand-pvc-error");
+          };
+
+          scope.$on('$destroy', hideErrorNotifications);
+
+          modalInstance.result.then(function(updatedSize) {
+            // upon clicking edit button, edit resource and send notification
+            ProjectsService
+            .get($routeParams.project)
```

Comment:
I don't think you need to get the project. I'd just use `{ namespace: scope.pvc.metadata.namespace }` as the context.